### PR TITLE
Add LaTeX printing for matrix symbol trace

### LINKED ISSUE #15303: LaTeX printing for Matrix Expression
```py
>>> A = MatrixSymbol("A", n, n)
>>> latex(trace(A**2))
'Trace(A**2)'
```

The bad part is not only is Trace not recognized, but whatever printer is being used doesn't fallback to the LaTeX printer for the inner expression (it should be `A^2`).

### PR DESCRIPTION
**Pull request by ethankward**
Fixes #15303. Adds a _print_Trace method and test. Also adds a _print_Basic method to handle unimplemented types.

* printing
  * added a method to correctly LaTeX print the trace of a MatrixSymbol
  * fixed a bug in LaTeX.py where subexpressions of an unimplemented basic expression were not printed correctly

### CODE REVIEW

**Comment by asmeurer** [comment]
I guess the problem with the fallback is that `Trace` directly subclasses `Expr`, which isn't defined on LaTeX printer, so it falls back to the EmptyPrinter, which is just `str`. The solution is to define `_print_Basic` on `LatexPrinter` analogous to the version on `StrPrinter`.

File: sympy/printing/latex.py [hunk 1]
```diff
@@ -1462,6 +1462,10 @@ def _print_Transpose(self, expr):
         else:
             return "%s^T" % self._print(mat)
 
+    def _print_Trace(self, expr):
+        mat = expr.arg
+        return r"\operatorname{Tr} %s" % self._print(mat)
```
Comment (asmeurer) [inline sympy/printing/latex.py]: I think it would be better to use `\mathrm{Tr}\left (%s \right )`.

Comment (asmeurer) [reply sympy/printing/latex.py]: Should it be `tr` with a lowercase `t`? That's what the pretty printer uses, and it's how Wikipedia renders it https://en.wikipedia.org/wiki/Trace_(linear_algebra) (Wikipedia also uses the parentheses everywhere).

Comment (ethankward) [reply sympy/printing/latex.py]: OK, that makes sense to me, for consistency. (http://mathworld.wolfram.com/MatrixTrace.html capitalizes, so it seems fairly arbitrary).

File: sympy/printing/latex.py [hunk 2]
```diff
@@ -289,6 +289,10 @@ def _do_exponent(self, expr, exp):
         else:
             return expr
 
+    def _print_Basic(self, expr):
+        l = [self._print(o) for o in expr.args]
+        return expr.__class__.__name__ + "(%s)" % ", ".join(l)
```
Comment (asmeurer) [inline sympy/printing/latex.py]: It should latexify the class name (I think the function you want is `self._deal_with_super_sub`).

File: sympy/printing/tests/test_latex.py [hunk 3]
```diff
@@ -1845,3 +1845,28 @@ def test_latex_printer_tensor():
...
+    class UnimplementedExpr(Expr):
+        def __new__(cls, e):
+            return Basic.__new__(cls, e)
+
+    def unimplemented_expr(expr):
+        # dummy function for testing
+        return UnimplementedExpr(expr).doit()
+
+    assert latex(unimplemented_expr(x)) == r'UnimplementedExpr\left(x\right)'
+    assert latex(unimplemented_expr(x**2)) == r'UnimplementedExpr\left(x^{2}\right)'
```
Comment (asmeurer) [inline sympy/printing/tests/test_latex.py]: Add a check here that the super_sub replacement is done correctly.

**Comment by asmeurer** [comment]
I have just the one comment.

**Pull request merged**